sencha-workspace/packages/slate-cbl/src/view/student/Dashboard.js

```diff
@@ -0,0 +1,197 @@
+                '<ul class="cbl-skill-demos">',
+                    '<tpl for="this.getDemonstrationBlocks(values)">',
+                        '<li class="cbl-skill-demo <tpl if="!counted">cbl-skill-demo-uncounted</tpl>" data-skill="{SkillID}" <tpl if="DemonstrationID">data-demonstration="{DemonstrationID}"</tpl>>',
+                            '<tpl if="Level==0">M<tpl else>{Level:defaultValue("&nbsp;")}</tpl>',
+                        '</li>',
+                    '</tpl>',
+                '</ul>',
+                '<div class="cbl-skill-description"><p>{Statement}</p></div>',
+//                '<div class="cbl-skill-complete-indicator cbl-level-{parent.level} is-checked">',
+//                    '<svg class="check-mark-image" width="16" height="16">',
+//                        '<polygon class="check-mark" points="13.824,2.043 5.869,9.997 1.975,6.104 0,8.079 5.922,14.001 15.852,4.07"/>',
+//                    '</svg>',
+//                '</div>',
+            '</li>',
+        '</tpl>',
+        {
+            // TODO: deduplicate this and its copy from the teacher dashboard?
+            getDemonstrationBlocks: function(skill) {
+                var demonstrationsRequired = skill.DemonstrationsRequired,
+                    blocks, blocksLength, blockIndex, lowestBlockIndex;
+
+
+                // start with all demonstrations 1 level below the competency level or greater
+                blocks = Ext.Array.map(skill.demonstrations, function(demonstration) {
+                    return Ext.apply({
+                        counted: demonstration.Level >= 8 // TODO: retrieve the competency level dynamically rather than hard coding to 9
+                    }, demonstration);
+                });
+
+
+                // trim lowest demonstrations
+                while ((blocksLength = blocks.length) > demonstrationsRequired) {
+                    for (blockIndex = 0, lowestBlockIndex = null; blockIndex < blocksLength; blockIndex++) {
+                        if (lowestBlockIndex === null || blocks[blockIndex].Level < blocks[lowestBlockIndex].Level) {
+                            lowestBlockIndex = blockIndex;
+                        }
+                    }
+
+                    Ext.Array.splice(blocks, lowestBlockIndex, 1);
+                }
+                
+                // sort counted demonstrations first
+                Ext.Array.sort(blocks, function(a, b) {
+                    if (a.counted && !b.counted) {
+                        return -1;
+                    }
+
+                    if (!a.counted && b.counted) {
+                        return 1;
+                    }
+
+                    return a.DemonstrationID > b.DemonstrationID ? 1 : -1;
+                });
+
+
+                // add empty blocks
+                while (blocks.length < demonstrationsRequired) {
+                    blocks.push({});
+                }
+                
+                if(blocks[0] == {}) {
```

Comment:
This isn't a good way of checking for an empty object. It looks like you can use ``Ext.isEmpty`` instead.